ui/plugin-system/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx

```diff
@@ -106,38 +120,20 @@ export function VariableEditForm(props: VariableEditFormProps) {
       >
         <Typography variant="h2">{title}</Typography>
         <Stack direction="row" spacing={1} sx={{ marginLeft: 'auto' }}>
-          {action === 'read' ? (
+          {action !== 'read' && (
             <Button
-              color="secondary"
-              variant="outlined"
+              disabled={!validation.isValid}
+              variant="contained"
               onClick={() => {
-                onCancel();
+                onChange(getVariableDefinitionFromState(state));
```

Comment:
From looking at code quickly these changes look good, I do think the feature branch idea [here](https://github.com/perses/perses/pull/1256/#discussion_r1264126731) might be the way to go since there are more PRs coming and to catch potential bugs before they get to `main`. To be safe, I'll probably end up doing a [snapshot release](https://github.com/perses/perses/blob/main/RELEASE.md#how-to-cut-a-ui-snapshot-release) off that feature branch so I can confirm all the plugin-system and dashboards package changes look good together in the Chronosphere app since there are some new props and breaking changes to components